pkg/ipmi/ipmi.go

```diff
@@ -75,50 +78,34 @@ const (
 var (
 	_IPMICTL_RECEIVE_MSG  = ioctl.IOWR(_IPMI_IOC_MAGIC, 12, uintptr(unsafe.Sizeof(recv{})))
 	_IPMICTL_SEND_COMMAND = ioctl.IOR(_IPMI_IOC_MAGIC, 13, uintptr(unsafe.Sizeof(req{})))
+
+	timeout = 30 * time.Second
 )
 
 type IPMI struct {
 	*os.File
 }
 
-type msg struct {
-	netfn   byte
-	cmd     byte
-	dataLen uint16
-	data    unsafe.Pointer
+type Msg struct {
+	Netfn   byte
+	Cmd     byte
+	DataLen uint16
```

Comment:
This is pretty nice!

I think it'd be nice if the user just had to pass a []byte at this point and pkg/IPMI had an internal version where it filled the datalen and pointer. Wdyt?

Reply:
I think that'd be awesome! The reason I didn't change the existing ones was because they all specify the netfn and cmd, and I think some functions rely on being able to send the datalen (see ClearCMOSClearValidBits - although I may have misunderstood that one).

So I just added a new function `SendRecvBasic` that does that without changing the current functions - is that ok?